src/machine.ts

```diff
@@ -86,13 +146,46 @@ export const SyncMachine = Machine<SyncContext, SyncSchema>({
   initial: 'idle',
   context: {
     hasMoreBlocks: false,
+    hasMempoolUpdate: false,
   },
   states: {
     idle: {
       always: [
         { target: 'syncing', cond: 'hasMoreBlocks' },
+        { target: 'mempoolsync', cond: 'hasMempoolUpdate' },
```

Comment:
Add a comment explaining that if both conditions are true, it will target the first one (syncing) as per https://xstate.js.org/docs/guides/guards.html#multiple-guards

Reply:
Done.